string directoryPath
---
bool success
string error